qiskit_ibm_provider/utils/backend_converter.py

```diff
@@ -0,0 +1,142 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Converters for migration from IBM Quantum BackendV1 to BackendV2."""
+
+from typing import Any, Dict
+
+from qiskit.transpiler.target import Target, InstructionProperties
+from qiskit.providers.backend import QubitProperties
+from qiskit.utils.units import apply_prefix
+from qiskit.circuit.library.standard_gates import IGate, SXGate, XGate, CXGate, RZGate
+from qiskit.circuit.parameter import Parameter
+from qiskit.circuit.gate import Gate
+from qiskit.circuit.measure import Measure
+from qiskit.circuit.reset import Reset
+from qiskit.providers.models import (
+    BackendConfiguration,
+    BackendProperties,
+    PulseDefaults,
+)
+
+
+def convert_to_target(
```

Comment:
There is a copy of this in terra but under test utils which we can't rely upon https://github.com/Qiskit/qiskit-terra/blob/5ed5db90b7f4b5988647aa72384b7833ea1184a7/qiskit/test/mock/utils/backend_converter.py

@mtreinish Do you think this can be moved to terra and consumed by providers? (I don't have a strong opinion either way).

Reply:
The primary reason I didn't do this directly in terra is I really want to be rid of the `BackendConfiguration`, `BackendProperties`, and `PulseDefaults` objects. They're poorly designed, IBM specific (while terra is supposed to be vendor agnostic), and hard to use. When I added BackendV2 it was in a big part to move away from these objects and I was reluctant to have anything that leverages these backend model objects as part of that.

The second more minor detail is the implementation currently in terra for the fake backends and what was used in the runtime client are subtlety different. The copy in terra's fake backend is closer to what I originally wrote for this converter function and works directly with the parsed json payload dictionaries and goes straight from dict -> Target. While IIRC for the runtime provider @rathishcholarajan adapted it to work with the model objects instead of the raw dictionary.

That being said I think that I agree and we probably will want to add these utility function to terra in the medium term despite my hesitations because I think for terra internally we're going to want to start using the `Target` directly as the sole payload we pass to transpiler passes (instead of as an optional kwarg like it's done today). So having a way for terra to generate a Target from a BackendV1 object (with the model objects) as the first step in `transpile()`. I think for terra 0.21 or 0.22 that would be a very reasonable thing to add, but for the near term this needs to be defined somewhere for the ibm provider package to use.

Reply:
I guess the challenge is that our current core package (terra) is not supposed to have IBM specific code. But we now have 3 different IBM specific packages with common code requirements. The lack of an IBM specific core package might be the source of some of the discussions of code duplication we had lately.

For this particular case I think we're good as the mid-term plan is to add shared utility functions to terra. Fine with me to live with duplication short-term.